include/oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h

```diff
@@ -0,0 +1,31 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_SCORING_POLICY_DEFS_H
+#define _ONEDPL_SCORING_POLICY_DEFS_H
+
+#include "oneapi/dpl/internal/dynamic_selection_traits.h"
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+  template<typename Policy, typename Resource>
+  class basic_selection_handle_t {
+    Policy p_;
+    Resource e_;
+  public:
+    explicit basic_selection_handle_t(const Policy &p, Resource e = Resource{}) : p_(p), e_(e) {}
```

Comment:
`std::move(e)` might be more efficient even for "Common Reference" types if they are written based on shared_ptr or something

Reply:
Will do later.